pretty_midi/pretty_midi.py

```diff
@@ -402,6 +403,161 @@ def estimate_tempo(self):
         """
         return self.estimate_tempi()[0][0]
 
+    def get_beats_using_metadata(self):
+        """Uses Time Signature and Tempo metadata to estimate beat times
+
+        Returns
+        -------
+        beat_times : np.ndarray
+            List of beat times in seconds
+        """
+
+        # threshold in seconds for merging beat locations
+        epsilon = 0.05
+
+        # get qpms (quarter per minute) tempi and their change time
+        tempo_change_times, qpms = self.get_tempo_changes()
+
+        # container for storing beat times
+        beat_times = []
+
+        # store time signature objects
+        ts_idx = 1
+        if len(self.time_signature_changes) > 1:
+            cur_ts = self.time_signature_changes[0]
+            nxt_ts = self.time_signature_changes[1]
+        else:
+            cur_ts = self.time_signature_changes[0]
+            nxt_ts = None
+
+        # go over tempo change times and time signatures assuming that the first
+        # tempo change time is the start time of the first beat
+        i = 1
+        beat_start = tempo_change_times[0]
+        while i < len(tempo_change_times):
+            # create beat times by going over time signatures while the next
+            # time signature happens before the next tempo change. note that
+            # time signatures always begin on the start of a beat
+            while nxt_ts is not None and nxt_ts.time < tempo_change_times[i]:
+                # update bpm given current time signature and qpm
+                bpm = qpm_to_bpm(qpms[i-1],
+                                 cur_ts.numerator,
+                                 cur_ts.denominator)
+
+                # interpolate from cur to nxt time signature change time
+                beat_times = np.hstack((beat_times, np.arange(beat_start,
```

Comment:
Ah, I see, so you're using np.arange as "accumulate until you have equalled or exceeded this point".  What do you want it to do when `start + n*step` is epsilon-close to `nxt_tempo_change`?  I'd rather this be handled explicitly, because `np.arange` is a good way to not be explicit about it.  For example, to ensure that `nxt_tempo_change` is never included, you could do something like 
```Python
new_beats = [beat_start]
while (new_beats[-1] < nxt_tempo_change and
       not np.allclose(new_beats[-1], nxt_tempo_change)):
    new_beats.append(new_beats[-1] + 60./bpm)
new_beats = new_beats[-1]
beat_times = np.append(beat_times, new_beats)
```